autotx/utils/ethereum/uniswap/swap.py

```diff
@@ -27,18 +29,19 @@ def get_swap_information(
     token_out_decimals: int,
     price: float,
     exact_input: bool,
+    web3: Web3
 ):
     if exact_input:
         amount_compared_with_token = amount * price
-        minimum_amount_out = int(amount_compared_with_token * 10**token_out_decimals)
-        amount_out = int(minimum_amount_out - (minimum_amount_out * SLIPPAGE))
-        return (amount_out, int(amount * 10**token_in_decimals), "exactInputSingle")
+        minimum_amount_out = int(Decimal(amount_compared_with_token) * 10**token_out_decimals)
```

Comment:
These unit conversions between decimal & int seems strange to me. I think we only need decimal in the first step where we multiple `amount * price`, and then we can simply `int(amount_compared_with_token * 10**token_out_decimals)` which moves us back into the integer w/ decimals space.

Reply:
Would revisit the rest of these modifications and do them only in the place where you're needed fractional numbers. Ideally this would be in a separate PR as it's sensitive.

Reply:
We can't do this, amount_compared_with_token is a float and if multiplied by a large number (like 10**token_out_decimals) it will have floating point precision errors.
![image](https://github.com/polywrap/AutoTx/assets/12792576/676d4e1a-10a1-4a1c-a9ce-db7088870971)